src/main/java/com/example/SpringExample/helper/MyDBHealthService.java

```diff
@@ -0,0 +1,26 @@
+package com.example.SpringExample.helper;
+
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Controller;
+import org.springframework.stereotype.Service;
+
+@Service
+public class MyDBHealthService implements HealthIndicator {
+
+    public static final String DB_SERVICE = "Database Service";
+
+    public boolean checkHealth(){
+        return true;
+    }
+
+    @Override
+    public Health health() {
+        if (checkHealth()) {
+            return Health.up().withDetail(DB_SERVICE, "Databse Sevice is running").build();
+        }
+        return Health.down().withDetail(DB_SERVICE, "Database service is running down").build();
+    }
```

Comment:
```suggestion
    @Override
    public Health health() {
        try (Connection conn = dataSource.getConnection()) {
            if (conn.isValid(1000)) {
                return Health.up().withDetail(DB_SERVICE, "Database Service is running").build();
            }
        } catch (SQLException e) {
            return Health.down().withDetail(DB_SERVICE, "Database Service is down: " + e.getMessage()).build();
        }
        return Health.down().withDetail(DB_SERVICE, "Database Service is unreachable").build();
    }
```

- The MyDBHealthService class does not really check the actual health of the database. Try suggestion above to implement it.

- The other possible way is by stopping the sql server in process settings.